File: String.cs
using System;

public class String
{
    private string data;

    public String()
    {
        data = "";
    }

    public String(string str)
    {
        data = str;
    }

    public String(String other)
    {
        data = other.data;
    }

    public int GetLength()
    {
        return data.Length;
    }

    public static String operator +(String s1, String s2)
    {
        return new String(s1.data + s2.data);
    }

    public static String operator *(String s, int n)
    {
        string newData = "";
        foreach (char c in s.data)
        {
            newData += new string(c, n);
        }
        return new String(newData);
    }

    public string GetData()
    {
        return data;
    }
}

File: Lab_1.3/String.cpp
#include "String.h"

String::String() : data("") {}

String::String(const std::string& str) : data(str) {}

String::String(const String& other) : data(other.data) {}

String::~String() {}

int String::getLength() const {
    return data.length();
}

String String::operator+(const String& other) const {
    return String(data + other.data);
}

String String::operator*(int n) const {
    std::string newData;
    for (char c : data) {
        newData += std::string(n, c);
    }
    return String(newData);
}

std::string String::getData() const {
    return data;
}
